File: tests/fixtures/jedi_project/services.py
from models import User, Product


def create_user(name: str, email: str) -> User:
    """Create and save a new user."""
    user = User(name=name, email=email)
    user.save()
    return user


def delete_user(user: User) -> bool:
    """Delete an existing user."""
    return user.delete()


def create_product(title: str, price: float) -> Product:
    """Create and save a new product."""
    product = Product(title=title, price=price)
    product.save()
    return product

File: tests/fixtures/ts_project/src/services.ts
import { User, Product, BaseService } from "./models";

/** Manages user operations. */
export class UserService extends BaseService {
  /** Create a new user. */
  async createUser(name: string, email: string): Promise<User> {
    this.log(`Creating user: ${name}`);
    return { id: 1, name, email };
  }

  /** Delete a user by ID. */
  async deleteUser(id: number): Promise<boolean> {
    return true;
  }
}

/** Fetch a user by ID. */
export const getUser = async (id: number): Promise<User | null> => {
  return null;
};

/** Create a product. */
export const createProduct = (title: string, price: number): Product => {
  return { id: 1, title, price };
};
